Automattic/wp-calypso
# Fixes the badge count for Promote post page in Woo

With the latest changes we are making in the Blaze Dashboard (promote post), we broke the badges we show in that page when you load the page. This happens because for Woo, we fetch only products, and the previous logic was assuming that it was unfiltered.

The idea of this PR is to add a second fetch to the same endpoint, but unfiltered and only querying for a single post (for performance reasons). With this, we can use the result from that call to get the correct total value for the badges.

![Screenshot 2024-01-17 at 11 12 51](https://github.com/Automattic/wp-calypso/assets/1258162/aa3f7612-06ce-44fa-bf74-3dc3e6847ac3)

## Proposed Changes

* Add a new hook that will be in charge of getting the correct total count of posts for the site

## Testing Instructions

* Open the live preview and navigate to Tools->Advertising
* Open the network console and filter by `/blaze/posts`
* Refresh the page and verify that we are calling the `/blaze.posts` twice, but one time only getting a single post:
![Screenshot 2024-01-17 at 11 19 15](https://github.com/Automattic/wp-calypso/assets/1258162/6e500324-c952-461d-a702-80f3f28b8887)
* Verify that the post count badge is correct, and shows the total number of posts for that site
* Switch to a site that doesn't have any products (not a WooCommerce site)
* Append this flag to the URL (`?flags=is_running_in_woo_site`) to see the Woo Blaze version of the Dashbaord.
* Again, verify that you see the correct post count in the badge, even if you don't have any products in the table

## Pre-merge Checklist
